hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/UpdateHoodieTableCommand.scala

```diff
@@ -56,7 +56,9 @@ case class UpdateHoodieTableCommand(ut: UpdateTable) extends HoodieLeafRunnableC
     val condition = ut.condition.getOrElse(TrueLiteral)
     val filteredPlan = Filter(condition, Project(targetExprs, ut.table))
 
-    val config = buildHoodieConfig(catalogTable)
+    // Set config to show that this is a prepped write.
+    var config = buildHoodieConfig(catalogTable)
```

Comment:
sorry. may be I might have mis-guided you. lets try to use "val" as much as possible to avoid un-intentional updates down the line. 
can you introduce a new "val" and use that below.

Reply:
Fixed.